contrib/repro_test.bzl

```diff
@@ -0,0 +1,261 @@
+        "history",
+        "file",
+        "size",
+        "rpm",
+        "apt",
+        "pip",
+        "node",
+    ],
+    image = "//testdata:derivative_with_volume",
+    workspace_file = "//:WORKSPACE",
+)
+"""
+
+load("@bazel_skylib//lib:dicts.bzl", "dicts")
+load("//container:container.bzl", _container = "container")
+load("//docker/util:run.bzl", _extract = "extract")
+
+def _impl(ctx):
+    """Core implementation of container_repro_test"""
+
+    name = ctx.attr.name
+    workspace_file = ctx.file.workspace_file
+    proj_root = ctx.actions.declare_file(name + "_src_project_root.txt")
+
+    # Record the absolute path to the provided WORKSPACE file and assume it to
+    # be the root of the Bazel project to build the 'image' in.
+    ctx.actions.run_shell(
+        command = "readlink -f %s | xargs dirname >> %s" % (workspace_file.path, proj_root.path),
+        inputs = [workspace_file],
+        outputs = [proj_root],
+    )
+
+    # Tar the current source project files to mount into the image that will
+    # be used to build and repro the 'image' in.
+    proj_tar = ctx.actions.declare_file(name + "_src_project.tar")
+    ctx.actions.run_shell(
+        command = "tar -cvf %s \"$(cat %s)\"" % (proj_tar.path, proj_root.path),
+        inputs = [proj_root],
+        outputs = [proj_tar],
+    )
+
+    # TODO(alex1545): It's a good idea to remove this big file after the test
+    # is executed, especially that it's only needed as an intermediate step.
+    # Try to find out if there is a way to remove a declared file.
+    # Simply running ctx.actions.run_shell with the 'rm' commond doesn't work.
+    image_tar = ctx.actions.declare_file(name + ".tar")
+
+    # Build the base image to build and repro the 'image' in.
+    # Mount the source project and its path into this image.
+    _container.image.implementation(
+        ctx,
+        files = [proj_root],
+        tars = [proj_tar],
+        output_executable = ctx.actions.declare_file(name + "_load.sh"),
+        output_tarball = image_tar,
+        workdir = "/",
+    )
+
+    img_label = ctx.attr.image.label
+    img_name = img_label.name
+    img_pkg = img_label.package
+    img_target_str = "//" + img_pkg + ":" + img_name
+
+    build_targets = [
+        img_target_str,
+        img_target_str + ".tar",
+        img_target_str + ".digest",
+        img_target_str + ".json",
+    ]
+    extract_path = "/img_outs"
+    img_tar = img_name + ".tar"
+    img_digest = img_name + ".digest"
+    commands = [
+        "cd \$(cat /%s)" % proj_root.basename,
+        "bazel build " + " ".join(build_targets),
+        "mkdir %s" % extract_path,
+        "cp bazel-bin/%s/%s %s/%s" % (img_pkg, img_tar, extract_path, img_tar),
+        "cp bazel-bin/%s/%s %s/%s" % (img_pkg, img_digest, extract_path, img_digest),
+        "cp \$(readlink -f bazel-bin/%s/%s).sha256 %s/%s" % (img_pkg, img_name + ".json", extract_path, img_name + ".id"),
+    ]
+    docker_run_flags = [
```

Comment:
add short doc above indicating -v is needed for docker sibling

Reply:
Done